Client/Experiments/Messaging/GleanPlumbMessageManager.swift

```diff
@@ -88,39 +90,78 @@ class GleanPlumbMessageManager: GleanPlumbMessageManagerProtocol {
     /// Perform any startup setup if necessary.
     func onStartup() { }
 
-    func hasMessage(for surface: MessageSurfaceId) -> Bool {
-        return getNextMessage(for: surface) != nil
-    }
-
     /// Returns the next valid and triggered message for the surface, if one exists.
     func getNextMessage(for surface: MessageSurfaceId) -> GleanPlumbMessage? {
         // All these are non-expired, well formed, and descending priority ordered messages for a requested surface.
-        let feature = messagingFeature.value()
-        let messages = getMessages(feature)
-            .filter {
-                $0.data.surface == surface
-            }
+        return getNextMessage(for: surface, availableMessages: getMessages(messagingFeature.value()))
+    }
 
+    func getNextMessage(for surface: MessageSurfaceId, availableMessages: [GleanPlumbMessage]) -> GleanPlumbMessage? {
         // If `NimbusMessagingHelper` creation fails, we cannot continue with this
         // feature! For that reason, return `nil`. We need to recreate the helper
         // for each request to get a message because device context can change.
         guard let messagingHelper = helperUtility.createNimbusMessagingHelper() else { return nil }
+        var jexlCache = [String: Bool]()
 
-        // Take the first triggered message.
-        guard let message = getNextTriggeredMessage(messages, messagingHelper) else { return nil }
+        var excluded: Set<String> = []
+        return getNextMessage(for: surface,
+                              availableMessages: availableMessages,
+                              excluded: &excluded,
+                              messagingHelper: messagingHelper,
+                              jexlCache: &jexlCache)
+    }
+
+    func getNextMessage(
```

Comment:
```suggestion
    // TODO: inout removal ticket https://mozilla-hub.atlassian.net/browse/FXIOS-6572
    func getNextMessage(
```
I have a ticket to eventually go in and remove inout because idempotence is a good thing. Adding this to do to keep track of it being here as well.